assembly/systemCalls.ts

```diff
@@ -800,16 +871,33 @@ export namespace System {
 
   /**
    * Require authority for an account
-   * @param type type of authority required
    * @param account account to check
+   * @param type type of authority required. By default it uses contract_call
+   * @param data data to be passed. By default it uses operation args
+   * @param caller contract caller. By default it calls the function to get the caller
    * @throws revert the transaction if the account is not authorized
    * @example
    * ```ts
-   * System.requireAuthority(authority.authorization_type.transaction_application, Base58.decode('1DQzuCcTKacbs9GGScRTU1Hc8BsyARTPqe));
+   * // check contract call authority
+   * System.requireAuthority(Base58.decode('1DQzuCcTKacbs9GGScRTU1Hc8BsyARTPqe));
+   *
+   * // if you already have the caller or args you can
+   * // pass them to save mana
+   * System.requireAuthority(
+   *   Base58.decode('1DQzuCcTKacbs9GGScRTU1Hc8BsyARTPqe),
+   *   authority.authorization_type.contract_call,
+   *   args,
+   *   caller
+   * );
    * ```
    */
-  export function requireAuthority(type: authority.authorization_type, account: Uint8Array): void {
-    require(checkAuthority(type, account), "account '" + Base58.encode(account) + "' authorization failed", error.error_code.authorization_failure);
+  export function requireAuthority(
+    account: Uint8Array,
```

Comment:
Again, let's keep the same interface for backwards compatibility.

Reply:
done